### ISSUE TITLE
The bio shows read more link when there is no content to show

### ISSUE THREAD

**Issue by murtazamalek** [exercism/exercism#6157]
I just created my profile on the website and set up my public profile. There is a link in the bio section which if there is no extra content to show, serves no purpose. It's minor but still can be fixed :)

Here is a quick video for reference: https://d.pr/v/anf6l4

**Comment by iHiD** [comment]
Thanks. Ideas on how to fix it?

**Comment by murtazamalek** [comment]
The most straightforward approach would be to check for the character length. It looks like the area allows 170 characters, so a simple check can be added in JS to match the length of the characters. If the characters are less than 170, hide the Read More link.

**Comment by iHiD** [comment]
How does that work on responsive screens where the number of characters that can fit on a line change?

**Comment by angelikatyborska** [comment]
@iHiD Just FYI it's a duplicate of a v3-beta issue that we didn't migrate here for some reason https://github.com/exercism/v3-beta/issues/15 and

**Comment by angelikatyborska** [comment]
> How does that work on responsive screens where the number of characters that can fit on a line change?

It doesn't. AFAIK if you want to cut content in a way that it gets revealed on a button press but don't have the button when it's not necessary, you cannot do that with CSS. You have to actually cut the content server-side or client-side with JS.

**Comment by iHiD** [comment]
Yeah - that's why I've not done it so far :) It's really not trivial.

**Comment by murtazamalek** [comment]
It's not trivial but still a user experience issue :) Please check the attached file. We can control the visibility of the characters via CSS width/height and a simple JS function to check for breakpoints and then show the button if needed.

To view the sample, rename Demo.txt and Demo.html. There are no dependencies, it's a simple JS function to be used as-is. 

[Demo.txt](https://github.com/exercism/exercism/files/7711451/Demo.txt)

**Comment by angelikatyborska** [comment]
Well, maybe it's not that bad to do. I implemented something similar at work not so long ago. It works like this:

1. Server-side, all content is render.
2. Client-side, a desired max-height is defined.
3. On page load, we check if the element's height (`el.getBoundingClientRect().height`) is more than the desired max height. If no, we just let it be. If yes, we add appropriate CSS to hide the excess content and add a button that would reveal it.
4. If you want it to be responsive, you need to repeat step 3. for window resize

**Comment by iHiD** [comment]
OK. If either of you want to PR it, then feel free :) 

The relevant React component is here; https://github.com/exercism/website/blob/main/app/javascript/components/common/Expander.tsx

You'll need to add a css file to the app/css/components. Current CSS (which might be able to be DRYd up) is in:
- https://github.com/exercism/website/blob/main/app/css/components/profile-header.css#L69-L81
- https://github.com/exercism/website/blob/main/app/css/pages/track-about.css#L86-L126

It'll need to be fully responsive (not breakpoints).

**Comment by angelikatyborska** [comment]
I'm not going to work on this after all. I cannot get https://github.com/exercism/development-environment to work and I don't really feel like spending hours on fighting with Docker 😞
